### PR TITLE
MISRA Directive 4.6 on MQTT Files

### PR DESCRIPTION
**Pull request by sarenameas**
*Issue #, if available:* https://issues.amazon.com/issues/TS-10693

*Description of changes:*
This covers all of the MISRA directive 4.6 violations on MQTT files and iot_logging.h and iot_logging.c. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: libraries/standard/mqtt/include/types/iot_mqtt_types.h [hunk 1]
```diff
@@ -35,6 +35,7 @@
 #include <stdbool.h>
 #include <stdint.h>
 #include <stddef.h>
+#include <inttypes.h>
```
Comment (sarenameas) [inline libraries/standard/mqtt/include/types/iot_mqtt_types.h]: Remove this and leave:
```
/* coverity[misra_c_2012_directive_4_6_violartion] */ 
```
on every violation line.

File: libraries/standard/mqtt/src/iot_mqtt_api.c [hunk 2]
```diff
@@ -845,10 +845,10 @@ bool _IotMqtt_IncrementConnectionReferences( _mqttConnection_t * pMqttConnection
     if( disconnected == false )
     {
         ( pMqttConnection->references )++;
-        IotLogDebug( "(MQTT connection %p) Reference count changed from %ld to %ld.",
+        IotLogDebug( "(MQTT connection %p) Reference count changed from %" PRId32 " to %" PRId32 ".",
```
Comment (sarenameas) [inline libraries/standard/mqtt/src/iot_mqtt_api.c]: Undo changes and annotate this line with:
```
/* coverity[misra_c_2012_directive_4_6_violartion] */ 
```

File: libraries/standard/mqtt/src/iot_mqtt_api.c [hunk 3]
```diff
@@ -845,6 +845,8 @@ bool _IotMqtt_IncrementConnectionReferences( _mqttConnection_t * pMqttConnection
     if( disconnected == false )
     {
         ( pMqttConnection->references )++;
+
+        /* coverity[misra_c_2012_directive_4_6_violation] */
```
Comment (abhidixi11) [inline libraries/standard/mqtt/src/iot_mqtt_api.c]: 'coverity' needs to be added in lexicon.txt file

Comment (gordonwang0) [reply libraries/standard/mqtt/src/iot_mqtt_api.c]: When adding annotations to disable MISRA rules for a line, please put a comment saying why.

Comment (sarenameas) [reply libraries/standard/mqtt/src/iot_mqtt_api.c]: Added to the atomics folder. Already present in mqtt.

**Comment by sarenameas** [comment]
Need to add IotLog() to MQTT lexicon.txt

**Comment by sarenameas** [comment]
This PR is ready for review

File: libraries/standard/mqtt/src/iot_mqtt_api.c [hunk 4]
```diff
@@ -910,6 +910,12 @@ bool _IotMqtt_IncrementConnectionReferences( _mqttConnection_t * pMqttConnection
     if( disconnected == false )
     {
         ( pMqttConnection->references )++;
+
+        /* In some implementations IotLog() maps to C standard printing API 
+         * that need specific primitive types for format specifiers. Also, 
+         * inttypes.h has may not available on some C99 compilers, despite
```
Comment (gkwicker) [inline libraries/standard/mqtt/src/iot_mqtt_api.c]: Need to fix this grammar – "Also, inttypes.h may not be available on some C99 compilers..."

File: libraries/standard/mqtt/src/iot_mqtt_api.c [hunk 5]
```diff
@@ -938,6 +944,11 @@ void _IotMqtt_DecrementConnectionReferences( _mqttConnection_t * pMqttConnection
     ( pMqttConnection->references )--;
     IotMqtt_Assert( pMqttConnection->references >= 0 );
 
+    /* In some implementations IotLog() maps to C standard printing API 
+     * that need specific primitive types for format specifiers. Also, 
+     * inttypes.h may not available on some C99 compilers, despite stdint.h
```
Comment (gkwicker) [inline libraries/standard/mqtt/src/iot_mqtt_api.c]: Needs a grammar fix, this appears throughout so please search/replace – "Also, inttypes.h may not be available on some C99 compilers..."

Comment (sarenameas) [reply libraries/standard/mqtt/src/iot_mqtt_api.c]: It is done. I will merge, I would like to squash commits.

**Pull request merged**